pkg/tsdb/cloudwatch/get_metric_query_batches_test.go

```diff
@@ -0,0 +1,146 @@
+		Id:               "m3",
+	}
+	math4 := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeRaw,
+		Expression:       "SUM(s1)",
+		Id:               "m4",
+	}
+	math5 := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeRaw,
+		Expression:       "PERIOD(i1) * RATE(i3)",
+		Id:               "m5",
+	}
+
+	t.Run("zero insight queries should not batch", func(t *testing.T) {
+		batch := []*models.CloudWatchQuery{
+			&metricStat,
+			&math1,
+			&math2,
+			&math3,
+			&math4,
+		}
+
+		result := getMetricQueryBatches(batch, logger)
+		assert.Len(t, result, 1)
+		assert.Equal(t, batch, result[0])
+	})
+
+	t.Run("one insight query should not batch", func(t *testing.T) {
+		batch := []*models.CloudWatchQuery{
+			&insight1,
+			&metricStat,
+		}
+
+		result := getMetricQueryBatches(batch, logger)
+		assert.Len(t, result, 1)
+		assert.ElementsMatch(t, batch, result[0])
+	})
+
+	t.Run("multiple insight queries should batch independent queries", func(t *testing.T) {
+		batch := []*models.CloudWatchQuery{
+			&insight1,
+			&metricStat,
+			&insight2,
+		}
+
+		result := getMetricQueryBatches(batch, logger)
+		assert.Len(t, result, 3)
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&insight1}, result[0])
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&metricStat}, result[1])
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&insight2}, result[2])
+	})
+
+	t.Run("math queries with one insight query should not batch", func(t *testing.T) {
+		batch := []*models.CloudWatchQuery{
+			&insight1,
+			&metricStat,
+			&math1,
+			&math2,
+			&math3,
+			&math4,
+		}
+
+		result := getMetricQueryBatches(batch, logger)
+		assert.Len(t, result, 1)
+		assert.ElementsMatch(t, batch, result[0])
+	})
+	t.Run("math queries with multiple insight queries should batch", func(t *testing.T) {
+		batch := []*models.CloudWatchQuery{
+			&insight1,
+			&insight2,
+			&metricStat,
+			&math1,
+			&math2,
+			&math3,
+			&math4,
+		}
+
+		result := getMetricQueryBatches(batch, logger)
+		assert.Len(t, result, 3)
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&insight2}, result[0])
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&insight1, &math1, &math2, &math3}, result[1])
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&metricStat, &math4}, result[2])
+	})
+	t.Run("a math query with multiple insight queries should get both", func(t *testing.T) {
+		batch := []*models.CloudWatchQuery{
+			&insight1,
+			&insight2,
+			&insight3,
+			&math1,
+			&math5,
+		}
+
+		result := getMetricQueryBatches(batch, logger)
+		assert.Len(t, result, 3)
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&insight2}, result[0])
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&insight1, &math1}, result[1])
+		assert.ElementsMatch(t, []*models.CloudWatchQuery{&insight1, &insight3, &math5}, result[2])
```

Comment:
should we comment that this third batch is expected to return an error from AWS?